### PR TITLE
Add CSV guess test

### PR DESCRIPTION
**Pull request by uu59**
Added CSV guess tests except `escape` option.
The reason is as [pending comment](https://github.com/embulk/embulk/pull/352/files#diff-3b3b0c90c6f748ef34ae306fd6c6f441R42), but I'm not sure that is intentional or bug...

### CODE REVIEW

**Comment by uu59** [comment]
@frsyuki do you have any idea for I described csv guess bahavior as above is intentional or not?

File: test/guess/test_csv_guess.rb [hunk 1]
```diff
@@ -0,0 +1,153 @@
+require 'helper'
+require 'time'
+require 'embulk/guess/csv'
+
+class CsvGuessTest < ::Test::Unit::TestCase
+  class TestDelimiter < self
+    data(
+      "\t" => "\t",
+      "," => "\t",
```
Comment (frsyuki) [inline test/guess/test_csv_guess.rb]: Do you mean "," here?

**Comment by frsyuki** [comment]
Looks very good. I commented one thing.
It would be great if you can add tests for corner cases such as `1, "value with space "" and quote in it "  ,"123",21150312T0000Z`

File: test/guess/test_csv_guess.rb (hunk 1 above)
Comment (uu59) [inline test/guess/test_csv_guess.rb]: Oops. thank you finding it!

**Comment by uu59** [comment]
Thanks review. I've added test for complex case.
How about `escape` option?

**Comment by frsyuki** [comment]
Looks good. I fixed `escape` at #354
